### PR TITLE
feat: Update Average Sale Price screen and My Collection artists list modal queries

### PR DESCRIPTION
**Pull request by MrSltun**
This PR resolves []

### Description
This PR updates the Average Sale Price route and screen to using the Artist ID to fetch the artist data instead of relying on the data from the previous screen's data

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [x] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- update average sale price and my collection artists list queries - mrsltun

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/Scenes/MyCollection/Screens/Insights/AverageSalePriceAtAuction.tsx [hunk 1]
```diff
@@ -1,24 +1,32 @@
...
-  collectorArtists,
+const AverageSalePriceAtAuctionScreen: React.FC<AverageSalePriceAtAuctionProps> = ({
+  refetch,
+  queryArgs,
 }) => {
   const [isVisible, setVisible] = useState<boolean>(false)
-  const [selectedArtist, setSelectedArtist] = useState<ArtistData>(artistData)
 
-  const enableChangeArtist = collectorArtists && collectorArtists > 1
+  const data = useLazyLoadQuery<AverageSalePriceAtAuctionQuery>(
```
Comment (lordkiz) [inline src/app/Scenes/MyCollection/Screens/Insights/AverageSalePriceAtAuction.tsx:21]: I thought @olerichter00 said we don't need to type these hooks anymore

File: src/app/Scenes/MyCollection/Screens/Insights/AverageSalePriceAtAuction.tsx [hunk 2]
```diff
@@ -1,24 +1,32 @@
...
+    AverageSalePriceAtAuctionScreenQuery,
+    queryArgs.variables,
+    queryArgs.options
+  )
+
+  const enableChangeArtist =
+    data.me?.myCollectionInfo?.artistsCount && data.me?.myCollectionInfo?.artistsCount > 0
```
Comment (lordkiz) [inline src/app/Scenes/MyCollection/Screens/Insights/AverageSalePriceAtAuction.tsx:28]:
```suggestion
  const enableChangeArtist = !!data.me?.myCollectionInfo?.artistsCount
```

File: src/app/Scenes/MyCollection/Screens/Insights/AverageSalePriceAtAuction.tsx [hunk 3]
```diff
@@ -64,13 +72,54 @@ export const AverageSalePriceAtAuction: React.FC<AverageSalePriceAtAuctionProps>
       </Flex>
 
       <AverageSalePriceSelectArtistModal
+        queryData={data}
         visible={isVisible}
         closeModal={() => setVisible(false)}
-        onItemPress={(artist) => {
-          setSelectedArtist(artist)
+        onItemPress={(artistId) => {
+          refetch(artistId)
```
Comment (lordkiz) [inline src/app/Scenes/MyCollection/Screens/Insights/AverageSalePriceAtAuction.tsx:79]: refetch? why?

File: src/app/Scenes/MyCollection/Screens/Insights/AverageSalePriceAtAuction.tsx (hunk 1 above)
Comment (MrSltun) [reply src/app/Scenes/MyCollection/Screens/Insights/AverageSalePriceAtAuction.tsx:21]: For `useLazyLoadQuery` I believe we still need it (need to confirm), but we don't need it anymore for `useFragment`

File: src/app/Scenes/MyCollection/Screens/Insights/AverageSalePriceAtAuction.tsx (hunk 3 above)
Comment (MrSltun) [reply src/app/Scenes/MyCollection/Screens/Insights/AverageSalePriceAtAuction.tsx:79]: no, it fetches the query with a different artist id when called, but maybe I should add a condition so it doesn't refetch when the same artist is selected 👍

File: src/app/Scenes/MyCollection/Screens/Insights/AverageSalePriceAtAuction.tests.tsx [hunk 4]
```diff
@@ -1,31 +1,51 @@
...
-  const mockArtist = {
-    name: "Andy Warhol",
-    formattedNationalityAndBirthday: "American, 1928–1987",
-    imageUrl: "https://d32dm0rphc51dk.cloudfront.net/E-k-uLoQADM8AjadsSKHrA/square.jpg",
-    initials: "AW",
-  }
+  const TestRenderer = () => {
+    useLazyLoadQuery<AverageSalePriceAtAuctionQuery>(AverageSalePriceAtAuctionScreenQuery, {
+      artistID: "artist-id",
+    })
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/Screens/Insights/AverageSalePriceAtAuction.tests.tsx:19]: 👏

**Review by olerichter00 — APPROVED** [review]
Looks great, thanks a lot 🥇

**Pull request merged**